internal/operator-controller/applier/helm.go

```diff
@@ -209,7 +213,119 @@ func (h *Helm) buildHelmChart(bundleFS fs.FS, ext *ocv1.ClusterExtension) (*char
 	if err != nil {
 		return nil, err
 	}
-	return h.BundleToHelmChartConverter.ToHelmChart(source.FromFS(bundleFS), ext.Spec.Namespace, watchNamespace)
+	var result *chart.Chart
+	contentType := bundleContentType(bundleFS)
+	switch contentType {
+	case HelmContent:
+		if !features.OperatorControllerFeatureGate.Enabled(features.HelmChartSupport) {
+			return nil, fmt.Errorf("helm chart support is not enabled")
+		}
+		result, err = loadChartWithEnrichments(bundleFS, WithInstallNamespace(ext.Spec.Namespace))
+		if err != nil {
+			return nil, fmt.Errorf("loading Helm chart; %w", err)
+		}
+	case BundleContent:
+		result, err = h.BundleToHelmChartConverter.ToHelmChart(source.FromFS(bundleFS), ext.Spec.Namespace, watchNamespace)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("unknown content found")
+	}
+
+	return result, nil
+}
+
+const (
+	HelmContent    string = "helm"
+	BundleContent  string = "bundle"
+	UnknownContent string = "unknown"
+)
+
+func bundleContentType(bundleFS fs.FS) string {
+	var contentType string
+	_ = fs.WalkDir(bundleFS, ".", func(path string, f fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !f.IsDir() {
+			// Check if Helm chart
+			if filepath.Dir(path) == "charts" &&
```

Comment:
The charts as far I understand is optional.
what is mandatory for any chart is 

Chart.yaml and dir templates/

So I think we want something like

```
func IsBundleSourceHelm(path string) bool {
	chartFile := filepath.Join(path, "Chart.yaml")
	templatesDir := filepath.Join(path, "templates")

	// Check if Chart.yaml file exists
	if _, err := os.Stat(chartFile); os.IsNotExist(err) {
		return false
	}

	// Check if templates directory exists and is a directory
	if stat, err := os.Stat(templatesDir); os.IsNotExist(err) || !stat.IsDir() {
		return false
	}

	return true
}
```

Reply:
The current implementation does not entail unpacking the contents of the Helm chart to the cache filesystem. However, if this is a hard requirement, then we can rewrite the implementation. For this reason, we have not taken the approach you have recommended above.

Reply:
Indeed there is room for simplifying the solution. 

With that being said, what is happening in the code today is, if we were pulling an imaginary Helm chart `registry.io/water-melon:1.0.0` from an OCI registry. It would stored in the bundleFS filesystem as `charts/water-melon-1.0.0.tgz`.

Which I can see was a poor naming choice. Going forward,  we have two options:

1) Change the name of the directory from `charts` to something else. e.g. `artifacts`
2) Alternatively, we can unpack the contents of the Helm chart archive to the filesystem to be able to inspect its contents.